src/main/scala/mesosphere/marathon/api/v2/validation/AppValidation.scala

```diff
@@ -304,7 +305,7 @@ trait AppValidation {
     }
   )
 
-  def validateCanonicalAppUpdateAPI(enabledFeatures: Set[String]): Validator[AppUpdate] = forAll(
+  def validateCanonicalAppUpdateAPI(enabledFeatures: Set[String], normalizationConfig: AppNormalization.Config): Validator[AppUpdate] = forAll(
```

Comment:
Trait works. Or else a func that produces an `Option[String]` (which would be even more minimal than a trait)